### PR TITLE
Improve the cmake files and format them

### PR DESCRIPTION
**Pull request by ClausKlein**
Added a cmake-format config file too.

Note: Do NOT export developer compile options like project-warnings and project-options!

### CODE REVIEW

File: .github/workflows/cmake.yml [hunk 1]
```diff
@@ -54,7 +55,7 @@ jobs:
             cxx: "g++",
           }
           - {
-            os: ubuntu-20.04,
+            os: ubuntu-22.04,
```
Comment (ClausKlein) [inline .github/workflows/cmake.yml:58]: With this version I have `ctest` errors like on OSX!
see https://github.com/ClausKlein/azmq/actions/runs/3554967099/jobs/5971384457

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,35 +1,46 @@
...
+    add_library(project_warnings INTERFACE)
+    if (MORE_WARNINGS)
+        include(cmake/CompilerWarnings.cmake)
+        set_project_warnings(project_warnings)
+    endif ()
+
+    add_library(project_options INTERFACE)
+    include(cmake/Sanitizers.cmake)
+    enable_sanitizers(project_options)
 endif ()
```
Comment (ClausKlein) [inline CMakeLists.txt]: This is for developers only, it must not installed!

File: CMakeLists.txt [hunk 3]
```diff
@@ -38,61 +49,44 @@ find_package(Threads REQUIRED)
...
-        INTERFACE
-        "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>"
-        "$<INSTALL_INTERFACE:include>"
-        )
-
-if (PROJECT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
-    set(CMAKE_MASTER_PROJECT YES)
+if (CMAKE_SKIP_INSTALL_RULES)
+    target_link_libraries(${PROJECT_NAME} INTERFACE project_options project_warnings)
 endif ()
```
Comment (ClausKlein) [inline CMakeLists.txt]: ... so the developer settings are not used if you want to install the library.

File: .cmake-format [hunk 4]
```diff
@@ -0,0 +1,16 @@
+format:
```
Comment (aboseley) [inline .cmake-format:1]: Nice, I haven't seen this before

File: .github/workflows/cmake.yml (hunk 1 above)
Comment (aboseley) [reply .github/workflows/cmake.yml:58]: I think this issue is fixed on master.  This branch is a bit out of date, I'll take a look at rebase it today

File: CMakeLists.txt (hunk 2 above)
Comment (aboseley) [reply CMakeLists.txt]: I agree it probably doesn't make sense to install the extra "build flag" targets that enable 'more warnings'.  Maybe these should get pushed down into the test project, so we only compile with these options when building the unit test.  Users of the project can the decide on their own flags.

**Comment by aboseley** [comment]
@ClausKlein ,  I've merged your changes, re-based on master, then reworked to give us a "AZMQ_DEVELOPER" option. Please let me know what you think.

**Pull request closed without merging**